### ISSUE TITLE
Segmentation Fault cudaMemcpy

### ISSUE THREAD

**Issue by tuero** [no-fee-ethereum-mining/nsfminer#73]
**Describe the bug**
After about an hour or so of running, a segmentation fault will be received and the program just hangs.

**To Reproduce**
1. Build from latest master source and install
2. Run and wait

**Expected behavior**
Program to not segfault

**Screenshots (Optional)**
```
 miner Job: 7d7f45dd daggerhashimoto.eu.nicehash.com:3353
 cu-0  Job: cdb21afe Solution: 0x27361600064e766a
 cu-0  Using 4.14 GB out of 23.70 GB GPU memory

Thread 7 "cu-0" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe5fff700 (LWP 16420)]
0x00007ffff51a77b0 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
(gdb) bt
#0  0x00007ffff51a77b0 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#1  0x00007ffff52a73ef in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#2  0x00007ffff539e1bd in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#3  0x00007ffff514ca0b in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#4  0x00007ffff53c7885 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#5  0x00007ffff5146337 in ?? () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#6  0x00007ffff51c2ed6 in cuMemcpyHtoD_v2 () from /usr/lib/x86_64-linux-gnu/libcuda.so.1
#7  0x00005555558f3599 in cudart::driverHelper::memcpyDispatch(void*, void const*, unsigned long, cudaMemcpyKind, bool) ()
#8  0x00005555558cd405 in cudart::cudaApiMemcpy(void*, void const*, unsigned long, cudaMemcpyKind) ()
#9  0x000055555590d3f9 in cudaMemcpy ()
#10 0x00005555558b6207 in dev::eth::CUDAMiner::initEpoch() ()
#11 0x00005555558b8419 in dev::eth::CUDAMiner::workLoop() ()
#12 0x0000555555670bb6 in dev::Worker::startWorking()::{lambda()#1}::operator()() const [clone .isra.40] ()
#13 0x00005555559692bf in execute_native_thread_routine ()
#14 0x00007ffff77b16db in start_thread (arg=0x7fffe5fff700) at pthread_create.c:463
#15 0x00007ffff6f24a3f in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Environment (please complete the following information):**
- Operating System: Ubuntu 18.04, CUDA 11.1
- Hardware: Nvidia 3090
- Ethminer Version: Latest master copy
- Ethminer options used: -P stratum2+tcp://ADDRESS@HOST:PORT -U

**Additional context**
Seems to run fine for ~ few hours, then crashes and hangs, without control being passed back.

**Comment by jean-m-cyr** [comment]
Can you provide the version as listed in the banner at startup?

**Comment by tuero** [comment]
```
 $nsfminer -P stratum2+tcp://ADDRESS@daggerhashimoto.eu.nicehash.com:3353 -U

 miner nsfminer 1.3.3 (No stinkin' fees edition)
 miner Copyright 2021 Jean M. Cyr, Licensed under the terms
 miner  of the GNU General Public License Version 3
 miner https://github.com/no-fee-ethereum-mining/nsfminer
 miner Build: linux/release/gnu
 miner 3rd Party: GCC 8.4.0, CUDA 11.1, Boost 1.75.0
 miner 3rd Party: OpenSSL 1.1.1i, Ethash 0.5.0
```

**Comment by angelir1** [comment]
Hello, in the Windows operating system, after several restarts of the program, video cards freeze, only restarting the computer helps.
 miner nsfminer 1.3.3 (No stinkin' fees edition)
 miner Copyright 2021 Jean M. Cyr, Licensed under the terms
 miner  of the GNU General Public License Version 3
 miner https://github.com/no-fee-ethereum-mining/nsfminer
 miner Build: windows/release/msvc
 miner 3rd Party: MSVC 19.16.27045.0, CUDA 11.2, Boost 1.75.0
 miner 3rd Party: OpenSSL 1.1.1i, Ethash 0.5.0
 miner Configured pool eth.2miners.com:2020
 <unknown> Selected pool eth.2miners.com:2020
 <unknown> Stratum mode : Eth-Proxy compatible
 <unknown> Established connection to eth.2miners.com:2020
 <unknown> Spinning up miners...
 cu-0  Using Pci 0000:01:00.0: GeForce RTX 3080 (Compute 8.6) Memory : 10.00 GB
 cu-1  Using Pci 0000:04:00.0: GeForce RTX 2080 (Compute 7.5) Memory : 8.00 GB
 cu-2  Using Pci 0000:05:00.0: GeForce RTX 2080 (Compute 7.5) Memory : 8.00 GB
 cu-3  Using Pci 0000:06:00.0: GeForce GTX 1080 (Compute 6.1) Memory : 8.00 GB

**Comment by jean-m-cyr** [comment]
I believe this is fixed in version 1.3.4. Reopen this issue if not.
